aiogram/types/input_file.py

```diff
@@ -45,6 +47,11 @@ def __init__(self, path_or_bytesio, filename=None, conf=None):
         elif isinstance(path_or_bytesio, _WebPipe):
             self._path = None
             self._file = path_or_bytesio
+
+        elif isinstance(path_or_bytesio, Path):
+            self._file = path_or_bytesio.open("rb")
+            self._path = path_or_bytesio.resolve()
+
```

Comment:
Filename should be resolved in the same way as it with string `path_or_bytesio`

```suggestion
            if filename is None:
                filename = path_or_bytesio.name
```